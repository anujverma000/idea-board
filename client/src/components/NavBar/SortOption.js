import React, { useState } from "react";
import styled from "styled-components";

const SortOption = ({ field, label, sort }) => {
  const [ sortOrder ,setSortOrder ] = useState('');

  const handleSort = () => {
    let type = sortOrder === 'desc' ? 'asc': 'desc'
    setSortOrder(type);
    sort({field, type})
  }
  return(
    <Option onClick={handleSort}>
      { label } 
      { sortOrder === '' && <DefaultSort/> }
      { sortOrder === 'asc' && <Up/> }
      { sortOrder === 'desc' && <Down/> }
    </Option>
  )
}

const Option = styled.div`
  font-size: 16px;
  text-align: center;
  cursor: pointer;
  padding: 12px;
  &:hover{
    background-color: ${props => props.theme.borderColor};  
  }
  border-bottom: 1px solid ${props => props.theme.borderColor};
  &:last-child{
    border-bottom: none;
  }
`
const DefaultSort = styled.div`
  width: 0;
  height: 0;
  float: right;
  margin: 4px;
  border-style: solid;
  border-width: 0 5px 8.7px 5px;
  border-color: transparent transparent ${props => props.theme.shadowColor} transparent;
`;

const Up = styled(DefaultSort)`
  border-width: 0 5px 8.7px 5px;
  border-color: transparent transparent ${props => props.theme.primaryColor} transparent;
`;

const Down = styled(DefaultSort)`
  border-width: 8.7px 5px 0 5px;
  border-color: ${props => props.theme.primaryColor} transparent transparent transparent;
`;
export default SortOption;