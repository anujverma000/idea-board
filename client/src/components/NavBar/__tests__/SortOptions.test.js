import React from 'react';
import renderer from 'react-test-renderer';
import SortOption from '../SortOption'

it('renders correctly', () => {
  const tree = renderer.create(<SortOption />).toJSON();
  expect(tree).toMatchSnapshot();
});
