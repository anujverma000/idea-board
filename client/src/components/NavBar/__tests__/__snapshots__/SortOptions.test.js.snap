// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="sc-bdVaJa fVLgYx"
  onClick={[Function]}
>
  <div
    className="sc-bwzfXH kvNVBS"
  />
</div>
`;
